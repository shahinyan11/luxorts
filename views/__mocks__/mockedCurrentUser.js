export default {
  id: '5c5d814c-8e06-433d-b006-4119311585bd',
  email: 'evhenii.hryhoriev+10@rubygarage.org',
  publicId: 'c4toxHVi',
  verifiedStatus: 'unverified',
  userType: 'traveller_account',
  userProfile: {
    id: '52080e19-c5e0-45bb-af58-edbbe70e78fa',
    about: null,
    address: null,
    avatarUrls: {
      original: '/avatar-img-url',
    },
    dateOfBirth: '1994-01-01',
    drivingLicenseFirstName: 'Evhenii',
    drivingLicenseLastName: 'Test',
    firstName: 'Evhenii',
    gender: null,
    lastName: 'Test',
    phoneNumber: '+299999999',
    street: 'street',
    city: 'city',
    state: 'state',
    zipCode: 'zipCode',
    country: 'country',
  },
  userInvitation: [
    {
      id: '52080e19-c5e0-45bb-af58-edbbe70e78fa',
      created_at: '2022-06-23T06:32:55.588Z',
      email: '12@mail.ru',
      last_sent_at: '2022-06-23T06:32:55.474Z',
      resendable: true,
      status: 'pending',
      user_account_id: '7211ccc2-1fca-4799-b1e1-4005e9fc42be',
    },
  ],
};
