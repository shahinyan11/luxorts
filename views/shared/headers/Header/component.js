import Link from 'next/link';
import PropTypes from 'prop-types';
import {FormattedMessage} from 'react-intl';
import {FormikProvider} from 'formik';
import {Form} from 'antd';
import classNames from 'classnames';
import Image from 'next/image';

import ROUTES from 'constants/routes';

import RangePickerField from 'views/shared/form/RangePickerField';
import GradientButton from 'views/shared/GradientButton';
import InputField from 'views/shared/form/InputField';
import SvgIcon from 'views/stubs/shared/SvgIcon';

import LoggedInUserWidget from './LoggedInUserWidget';
import LoggedOutUserWidget from './LoggedOutUserWidget';
import useContainer from './hook';

const Header = ({ isSignIn, isSignUp, isGuest, isWithFilter }) => {
  const {
    isLoggedInUserWidgetVisible,
    isLoggedOutUserWidgetVisible,
    formik,
    switchToHostingRoute,
  } = useContainer({
    isSignIn,
    isSignUp,
    isGuest,
  });

  return (
    <header className="main-header">
      <div
        className={classNames('main-header__container', {
          'main-header__container--filter': isWithFilter,
        })}
      >
        <Link href={ROUTES.INDEX.PATH}>
          <a className="main-header__logo">
            <Image src="/images/logo.svg" alt="" width="138" height="32" />
          </a>
        </Link>
        {isWithFilter && (
          <Form layout="vertical" size="large" className="search ml-xl-24 mr-xl-auto">
            <FormikProvider value={formik}>
              <div className="search__location search__location--header">
                <InputField
                  name="location"
                  placeholder={{ id: 'shared.location' }}
                  allowClear={{ clearIcon: <SvgIcon icon="cross" /> }}
                  className="search__input"
                  formItemClassName="search__item"
                />
              </div>
              <div className="search__dates search__dates--header">
                <RangePickerField
                  clearIcon={<SvgIcon icon="cross" />}
                  suffixIcon={false}
                  separator={false}
                  placeholder={[{ id: 'shared.checkIn' }, { id: 'shared.checkOut' }]}
                  className="w-100"
                  size="large"
                  name="date"
                />
              </div>
              <div className="search__guests search__guests--header">
                <InputField
                  name="guests"
                  placeholder={{ id: 'shared.guests' }}
                  allowClear={{ clearIcon: <SvgIcon icon="cross" /> }}
                  className="search__input"
                  formItemClassName="search__item mb-0"
                />
                <GradientButton
                  className="search__submit main-btn--small main-btn--icon"
                  variant="primary"
                  addonBefore={<SvgIcon icon="search" className="icon-right" />}
                />
              </div>
            </FormikProvider>
          </Form>
        )}
        {isSignIn && (
          <div className="d-flex flex-column flex-sm-row align-items-center">
            <span className="main-header__caption mb-8 mb-sm-0">
              <FormattedMessage id="shared.dontHaveAnAccountYet" />
            </span>
            <Link href={ROUTES.SIGN_UP.PATH}>
              <a>
                <GradientButton
                  className="main-btn--md main-header__button ml-sm-16 min-width-120"
                  text={{ id: 'shared.signUp' }}
                  variant="secondary"
                />
              </a>
            </Link>
          </div>
        )}
        {isSignUp && (
          <div className="d-flex flex-column flex-sm-row align-items-center">
            <span className="main-header__caption mb-8 mb-sm-0">
              <FormattedMessage id="shared.alreadyHaveAnAccount" />
            </span>
            <Link href={ROUTES.SIGN_IN.PATH}>
              <a>
                <GradientButton
                  className="main-btn--md main-header__button ml-sm-16 min-width-120"
                  text={{ id: 'shared.signIn' }}
                  variant="secondary"
                />
              </a>
            </Link>
          </div>
        )}
        {!isLoggedInUserWidgetVisible && (
          <LoggedInUserWidget switchToHostingRoute={switchToHostingRoute} />
        )}
        {isLoggedOutUserWidgetVisible && <LoggedOutUserWidget />}
      </div>
    </header>
  );
};

Header.defaultProps = {
  isSignIn: false,
  isSignUp: false,
  isGuest: false,
  isWithFilter: false,
};

Header.propTypes = {
  isSignIn: PropTypes.bool,
  isSignUp: PropTypes.bool,
  isGuest: PropTypes.bool,
  isWithFilter: PropTypes.bool,
};

export default Header;
