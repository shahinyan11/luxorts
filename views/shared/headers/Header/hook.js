import { useSelector } from 'react-redux';
import { useFormik } from 'formik';

import { USER_TYPE } from 'constants';
import ROUTES from 'constants/routes';

import isPresent from 'utils/isPresent';

import { currentUserSelector } from 'state/concepts/session/selectors';

function useContainer({ isSignIn, isSignUp, isGuest }) {
  const currentUser = useSelector(currentUserSelector);
  const isUserLoggedIn = isPresent(currentUser);

  const isAuthFlow = isSignIn || isSignUp || isGuest;
  const isLoggedOutUserWidgetVisible = !isAuthFlow && !isUserLoggedIn;
  const isLoggedInUserWidgetVisible = !isAuthFlow && isUserLoggedIn;

  const switchToHostingRoute =
    currentUser?.userType === USER_TYPE.TRAVELLER
      ? ROUTES.ADD_NEW_LISTING.INDEX.PATH
      : ROUTES.DASHBOARD.LISTINGS.PATH;

  /**
   * Formik initialization
   */
  const formik = useFormik({
    initialValues: {
      location: '',
      date: '',
      guests: '',
    },
  });

  return {
    isLoggedOutUserWidgetVisible,
    isLoggedInUserWidgetVisible,
    formik,
    switchToHostingRoute,
  };
}

export default useContainer;
