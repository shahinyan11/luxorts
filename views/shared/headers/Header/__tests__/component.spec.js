import { shallow } from 'enzyme';

import mockedCurrentUser from 'views/__mocks__/mockedCurrentUser';

import useContainer from '../hook';
import Header from '../component';

const mockedHookData = {
  isLoggedInUserWidgetVisible: false,
  isLoggedOutUserWidgetVisible: false,
  currentUser: mockedCurrentUser,
  onLogOutHandler: jest.fn(),
  switchToHostingRoute: 'switchToHostingRoute',
};
jest.mock('../hook', () => jest.fn(() => mockedHookData));

describe('Header component', () => {
  let component = null;

  beforeEach(() => {
    component = shallow(<Header />);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when isSignIn equals true', () => {
    component.setProps({
      isSignIn: true,
    });
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when isSignUp equals true', () => {
    component.setProps({
      isSignUp: true,
    });
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when isLoggedInUserWidgetVisible equals true', () => {
    useContainer.mockReturnValueOnce({
      ...mockedHookData,
      isLoggedInUserWidgetVisible: true,
    });
    component = shallow(<Header />);

    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when isLoggedOutUserWidgetVisible equals true', () => {
    useContainer.mockReturnValueOnce({
      ...mockedHookData,
      isLoggedOutUserWidgetVisible: true,
    });
    component = shallow(<Header />);

    expect(component).toMatchSnapshot();
  });
});
