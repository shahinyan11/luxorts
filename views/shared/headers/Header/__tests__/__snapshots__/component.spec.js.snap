// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<header
  className="main-header"
>
  <div
    className="main-header__container"
  >
    <Link
      href="/"
    >
      <a
        className="main-header__logo"
      >
        <Image
          alt=""
          height="32"
          src="/images/logo.svg"
          width="138"
        />
      </a>
    </Link>
  </div>
</header>
`;

exports[`Header component matches snapshot when isLoggedInUserWidgetVisible equals true 1`] = `
<header
  className="main-header"
>
  <div
    className="main-header__container"
  >
    <Link
      href="/"
    >
      <a
        className="main-header__logo"
      >
        <Image
          alt=""
          height="32"
          src="/images/logo.svg"
          width="138"
        />
      </a>
    </Link>
    <LoggedInUserWidget
      switchToHostingRoute="switchToHostingRoute"
    />
  </div>
</header>
`;

exports[`Header component matches snapshot when isLoggedOutUserWidgetVisible equals true 1`] = `
<header
  className="main-header"
>
  <div
    className="main-header__container"
  >
    <Link
      href="/"
    >
      <a
        className="main-header__logo"
      >
        <Image
          alt=""
          height="32"
          src="/images/logo.svg"
          width="138"
        />
      </a>
    </Link>
    <LoggedOutUserWidget />
  </div>
</header>
`;

exports[`Header component matches snapshot when isSignIn equals true 1`] = `
<header
  className="main-header"
>
  <div
    className="main-header__container"
  >
    <Link
      href="/"
    >
      <a
        className="main-header__logo"
      >
        <Image
          alt=""
          height="32"
          src="/images/logo.svg"
          width="138"
        />
      </a>
    </Link>
    <div
      className="d-flex flex-column flex-sm-row align-items-center"
    >
      <span
        className="main-header__caption mb-8 mb-sm-0"
      >
        <Memo(FormattedMessage)
          id="shared.dontHaveAnAccountYet"
        />
      </span>
      <Link
        href="/sign-up"
      >
        <a>
          <GradientButton
            addonAfter={null}
            addonBefore={null}
            className="main-btn--md main-header__button ml-sm-16 min-width-120"
            text={
              Object {
                "id": "shared.signUp",
              }
            }
            variant="secondary"
          />
        </a>
      </Link>
    </div>
  </div>
</header>
`;

exports[`Header component matches snapshot when isSignUp equals true 1`] = `
<header
  className="main-header"
>
  <div
    className="main-header__container"
  >
    <Link
      href="/"
    >
      <a
        className="main-header__logo"
      >
        <Image
          alt=""
          height="32"
          src="/images/logo.svg"
          width="138"
        />
      </a>
    </Link>
    <div
      className="d-flex flex-column flex-sm-row align-items-center"
    >
      <span
        className="main-header__caption mb-8 mb-sm-0"
      >
        <Memo(FormattedMessage)
          id="shared.alreadyHaveAnAccount"
        />
      </span>
      <Link
        href="/sign-in"
      >
        <a>
          <GradientButton
            addonAfter={null}
            addonBefore={null}
            className="main-btn--md main-header__button ml-sm-16 min-width-120"
            text={
              Object {
                "id": "shared.signIn",
              }
            }
            variant="secondary"
          />
        </a>
      </Link>
    </div>
  </div>
</header>
`;
